src/commands/force/org/shape/delete.ts

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) 2022, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { EOL } from 'os';
+import { flags, FlagsConfig, SfdxCommand } from '@salesforce/command';
+import { SfdxError, Messages, Connection } from '@salesforce/core';
+import { RecordResult } from 'jsforce';
+import { isShapeEnabled, JsForceError } from '../../../../shared/orgShapeListUtils';
+
+Messages.importMessagesDirectory(__dirname);
+const messages = Messages.loadMessages('@salesforce/plugin-signups', 'shape.delete');
+
+interface ShapeRepresentation {
+  Id: string;
+  Status: string;
+  Edition: string;
+  Features: string;
+  Settings: string;
+  CreatedDate: string;
+  Description: string;
+}
+
+export interface OrgShapeDeleteResult {
+  orgId: string;
+  shapeIds: string[];
+}
+export class OrgShapeDeleteCommand extends SfdxCommand {
+  public static readonly description = messages.getMessage('description');
+  public static readonly examples = messages.getMessage('help').split(EOL);
+  public static readonly requiresUsername = true;
+  public static readonly flagsConfig: FlagsConfig = {
+    noprompt: flags.boolean({
+      char: 'p',
+      description: messages.getMessage('noPrompt'),
+    }),
+  };
+  private conn: Connection;
+
+  public async run(): Promise<OrgShapeDeleteResult> {
+    if (!this.flags.noprompt) {
+      if (!(await this.ux.confirm(messages.getMessage('deleteCommandYesNo', [this.org.getUsername()])))) {
+        return;
+      }
+    }
+
+    this.conn = this.org.getConnection();
+
+    if (!(await isShapeEnabled(this.conn))) {
+      throw new SfdxError(messages.getMessage('noAccess', [this.org.getUsername()]));
+    }
+    const deletedShapesIds = await this.deleteAll();
+    if (deletedShapesIds.length === 0) {
+      this.ux.log(messages.getMessage('noShapesHumanSuccess', [this.org.getOrgId()]));
+      return;
```

Comment:
return if no shapes are found, current `shape:delete` command shows both error and succcess msg:
```
 sfdx force:org:shape:delete -u na40 --noprompt
Can't delete org shape. No org shape found for org 000000000000000.
Successfully deleted org shape for 000000000000000.
```